tests/exporter/terra/submission/test_handler.py

```diff
@@ -0,0 +1,68 @@
+import json
+import unittest
+import uuid
+from unittest.mock import MagicMock
+
+from kombu import Message
+
+from exporter.ingest.service import IngestService
+from exporter.session_context import get_session_value
+from exporter.terra.submission.exporter import TerraSubmissionExporter
+from exporter.terra.submission.handler import TerraSubmissionHandler
+
+
+class TestTerraSubmissionHandler(unittest.TestCase):
+    def setUp(self) -> None:
+        self.mock_exporter_start = MagicMock()
+        mock_exporter = MagicMock(spec=TerraSubmissionExporter)
+        mock_exporter.start_data_file_transfer = self.mock_exporter_start
+
+        self.mock_ingest_set = MagicMock()
+        mock_ingest = MagicMock(spec=IngestService)
+        mock_ingest.set_data_file_transfer = self.mock_ingest_set
+        self.handler = TerraSubmissionHandler(mock_exporter, mock_ingest)
+
+        self.msg_ack = MagicMock()
+        self.msg_reject = MagicMock()
+        self.message = MagicMock(spec=Message)
+        self.message.ack = self.msg_ack
+        self.message.reject = self.msg_reject
+
+    def test_handle_message(self):
+        # Given
+        export_job = str(uuid.uuid4())
+        submission = str(uuid.uuid4())
+        project = str(uuid.uuid4())
+        body = {
+            "exportJobId": export_job,
+            "submissionUuid": submission,
+            "projectUuid": project
+        }
```

Comment:
This is repeasted in the next function. HAve you considered a `make_message()` function that could be shared?

Reply:
I'd usually agree but this would end up with something like the following which I don't think this is better. Possibly because the required tuple return makes it noisy.
``` test.py
    def test_get_context(self):
        # Given
        export_job, submission, project, body = self.make_message_body()
        # When
        with self.handler.set_context(body):
            # Then
            self.assertEqual(submission, get_session_value('submission_uuid'))
            self.assertEqual(export_job, get_session_value('export_job_id'))
            self.assertEqual(project, get_session_value('project_uuid'))
        # And Then
        self.assertEqual('n/a', get_session_value('submission_uuid'))
        self.assertEqual('n/a', get_session_value('export_job_id'))
        self.assertEqual('n/a', get_session_value('project_uuid'))
    
    @staticmethod
    def make_message_body():
        export_job = str(uuid.uuid4())
        submission = str(uuid.uuid4())
        project = str(uuid.uuid4())
        return export_job, submission, project, {
            "exportJobId": export_job,
            "submissionUuid": submission,
            "projectUuid": project
        }
```